core/commands/name/name.go

```diff
@@ -62,5 +80,204 @@ Resolve the value of a dnslink:
 		"publish": PublishCmd,
 		"resolve": IpnsCmd,
 		"pubsub":  IpnsPubsubCmd,
+		"inspect": IpnsInspectCmd,
+	},
+}
+
+type IpnsInspectValidation struct {
+	Valid     bool
+	Reason    string
+	PublicKey peer.ID
+}
+
+// IpnsInspectEntry contains the deserialized values from an IPNS Entry:
+// https://github.com/ipfs/specs/blob/main/ipns/IPNS.md#record-serialization-format
+type IpnsInspectEntry struct {
+	Value        string
+	ValidityType *ipns_pb.IpnsEntry_ValidityType
+	Validity     *time.Time
+	Sequence     uint64
+	TTL          *uint64
+	PublicKey    string
+	SignatureV1  string
+	SignatureV2  string
+	Data         interface{}
+}
+
+type IpnsInspectResult struct {
+	Entry      IpnsInspectEntry
+	Validation *IpnsInspectValidation
+}
+
+var IpnsInspectCmd = &cmds.Command{
+	Status: cmds.Experimental,
```

Comment:
@hacdias (just fyi) good practice to add this for new commands, so it shows under Experimental section in https://docs.ipfs.tech/reference/kubo/rpc/ and allows us to change output in the future without stress ;)

Reply:
Thanks for the heads up!